### DISCUSSION TITLE
How can the model be saved for multiple times series?

### DISCUSSION THREAD

**Discussion by meetu30** [winedarksea/AutoTS#135] [Q&A]
Hi,
I am working on a problem which is like multiple products (say A,B,C,D,E) and their respective sales with week start as the index.
Now, I am trying to call AutoTS inside for-loop for each of the above products. This way, first, product A will go, get itself trained on all data points, and then predicts for the next say 12 weeks. Next, product B will go, and return me the forecasted sales for 12 weeks and so on. All 5 products have SAME number of rows, meaning start date and end date are all the same. Once I am done running AutoTS for 1 product, I save its respective models using export statement. This gives me top 6 models for product A, then again loop will run and I get another file for product B with top 6 models and so on. At the end, I get 5 csv files, 1 for each product and inside that csv I have top 6 models for each of the product. Then I run them again to pick 1 best model (this time I ask AutoTS to pick 1 from the top 6). My question is how can I deploy this? because in real I don't have 5 products I have them in 100s.  And it takes A LOT OF TIME , which is a major challenge here. because how will i deploy as I don't have any pickle file(tried saving the 1 best model and reading the same, but it was empty). Can you please guide? Thanks in anticipation.

**Comment by winedarksea** [comment]
Firstly, you shouldn't be calling AutoTS in a `for` loop. It is designed to handle being fed a bunch of products all at once. Check out some of the example datasets, but suffice to say I designed it specifically for forecasting on diverse multivariate product sets (with `horizontal` ensembles) all as one dataset.

Passing them all at once will solve many of your problems. As for the rest, while you can pickle an AutoTS model, I highly recommend at least limited re-training on each one (using the export_template, import_template calls). You can see an example of all this in the production_example.py of the repo.

Let me know if that helps @meetu30
